automates/text_reading/src/test/scala/org/clulab/aske/automates/text/TestFunctions.scala

```diff
@@ -96,31 +140,189 @@ class TestFunctions extends ExtractionTest {
+  val t2f = "Soil temperature is computed from air temperature and a deep soil temperature boundary condition that is calculated from the average annual air temperature and the amplitude of monthly mean temperatures."
+  passingTest should s"find functions from t2f: ${t2f}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "Soil temperature" -> Seq("air temperature", "deep soil temperature boundary condition"), // note: only one rule applies even when multiple rules should apply. needs to find out why.
+      "deep soil temperature boundary condition" -> Seq("average annual air temperature", "amplitude of monthly mean temperatures")
+    )
+    val mentions = extractMentions(t2f)
+    testFunctionEvent(mentions, desired)
+  }
+
+  val t3f = "This module brings together soil, plant and atmosphere inputs and computes light interception by the canopy, potential evapotranspiration (ET) as well as actual soil evaporation and plant transpiration"
+  failingTest should s"find functions from t3f: ${t3f}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "light interception" -> Seq("canopy", "potential evapotranspiration", "actual soil evaporation", "plant transpiration")
+    )
+    val mentions = extractMentions(t3f)
+    testFunctionEvent(mentions, desired)
+  }
+
+  val t4f = "Actual soil evaporation is the minimum of the potential and soil-limiting calculations on a daily basis."
+  passingTest should s"find functions from t4f: ${t4f}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "Actual soil evaporation" -> Seq("potential and soil-limiting calculations on a daily basis") // note: would input be one chunk? or two with conjunction?
+    )
+    val mentions = extractMentions(t4f)
+    testFunctionEvent(mentions, desired)
+  }
+
+  val t5f = "Photosynthesis of sunlit and shaded leaves is computed hourly using the asymptotic exponential response equation, " +
+    "where quantum efficiency and light-saturated photosynthesis rate variables are dependent on CO2 and temperature"
+  failingTest should s"find functions from t5f: ${t5f}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "quantum efficiency" -> Seq("CO2", "temperature"), // fixme: for now, function extraction is done only once.
+      "light-saturated photosynthesis rate variables" -> Seq("CO2", "temperature")
+    )
+    val mentions = extractMentions(t5f)
+    testFunctionEvent(mentions, desired)
+  }
 
+  val t6f = "Hourly canopy photosynthesis on a land area basis is computed from the sum of sunlit and shaded leaf contributions " +
+    "by multiplying sunlit and shaded leaf photosynthetic rates by their respective LAIs."
+  failingTest should s"find functions from t6f: ${t6f}" taggedAs(Somebody) in {
+    val desired = Seq(
+      "Hourly canopy photosynthesis on a land area basis" -> Seq("sunlit and shaded leaf contributions"),
+      "Hourly canopy photosynthesis on a land area basis" -> Seq("sunlit and shaded leaf photosynthetic rates", "their respective LAIs")
```

Comment:
Based on our discussion yesterday, should the desired extraction be a little more granular? like `sunlit leaf photosynthetic rates -> LAIs` and `sunlit leaf photosynthetic rates -> LAIs`?

Reply:
@alicekwak and @maxaalexeeva was this conversation ever resolved?

Reply:
@pauldhein yes, keeping it as is for the time being.